games/rogue/files/ncurses-compat-fix.patch

```diff
@@ -0,0 +1,75 @@
+diff -Nuar rogue5.4.4.orig/extern.h rogue5.4.4/extern.h
+--- rogue5.4.4.orig/extern.h	2007-09-06 04:08:06.000000000 +0900
++++ rogue5.4.4/extern.h	2024-11-24 10:16:48.684919316 +0900
+@@ -131,11 +131,11 @@
+ void	doctor();
+ void	end_line();
+ void    endit(int sig);
+-void	fatal();
++void	fatal(char *s);
+ void	getltchars();
+ void	land();
+ void    leave(int);
+-void	my_exit();
++void	my_exit(int st);
+ void	nohaste();
+ void	playit();
+ void    playltchars(void);
+@@ -144,7 +144,7 @@
+ void    resetltchars(void);
+ void	rollwand();
+ void	runners();
+-void	set_order();
++void	set_order(int*, int);
+ void	sight();
+ void	stomach();
+ void	swander();
+diff -Nuar rogue5.4.4.orig/main.c rogue5.4.4/main.c
+--- rogue5.4.4.orig/main.c	2007-09-06 04:08:06.000000000 +0900
++++ rogue5.4.4/main.c	2024-11-24 10:17:16.586566729 +0900
+@@ -238,8 +238,8 @@
+     getyx(curscr, y, x);
+     mvcur(y, x, oy, ox);
+     fflush(stdout);
+-    curscr->_cury = oy;
+-    curscr->_curx = ox;
++    oy = getcury(curscr);
++    ox = getcurx(curscr);
```

Comment:
This part looks wrong. The original code was storing the values of `oy` and `ox` into `curscr`, but now you're doing the opposite and storing the values from `curscr` into `oy` and `ox`.

Reply:
Oh! I'll check it again.

Reply:
@jmroot is right. But the reason there was no problem with the functional test was because these two lines were not doing anything functional. I'll fix patch and recommit.